lighty-examples/lighty-cluster-app/src/main/java/io/lighty/examples/controllers/cluster/Main.java

```diff
@@ -103,143 +115,114 @@ public void start(String[] args, boolean registerShutdownHook) {
-                restConfConfig.setInetAddress(InetAddress.getByName("0.0.0.0"));
+                restconfConfiguration = RestConfConfigUtils.getDefaultRestConfConfiguration();
+                restconfConfiguration.setWebSocketPort(restconfConfiguration.getWebSocketPort()
+                    + arguments.getMemberOrdinal());
+                restconfConfiguration.setHttpPort(restconfConfiguration.getHttpPort() + arguments.getMemberOrdinal());
+                restconfConfiguration.setInetAddress(InetAddress.getByName("0.0.0.0"));
                 //3. NETCONF SBP configuration
-                NetconfConfiguration netconfSBPConfig = NetconfConfigUtils.createDefaultNetconfConfiguration();
-                startLighty(defaultClusterNodeConfiguration, restConfConfig, netconfSBPConfig, registerShutdownHook);
+                netconfSBPConfiguration = NetconfConfigUtils.createDefaultNetconfConfiguration();
             }
-            float duration = (System.nanoTime() - startTime) / 1_000_000f;
-            LOG.info("lighty.io and RESTCONF-NETCONF started in {}ms", duration);
-        } catch (Throwable e) {
-            LOG.error("Main RESTCONF-NETCONF application exception: ", e);
+            startLighty(controllerConfiguration, restconfConfiguration, netconfSBPConfiguration, registerShutdownHook);
+            LOG.info("Lighty.io and RESTCONF-NETCONF started in {}", stopwatch.stop());
+        } catch (Throwable cause) {
+            LOG.error("Main RESTCONF-NETCONF application exception: ", cause);
+            shutdown();
         }
     }
 
-    private void startLighty(ControllerConfiguration controllerConfiguration,
-            RestConfConfiguration restConfConfiguration, NetconfConfiguration netconfSBPConfiguration,
-            boolean registerShutdownHook)
-            throws ConfigurationException, ExecutionException, InterruptedException {
+    private void startLighty(final ControllerConfiguration controllerConfiguration,
+                             final RestConfConfiguration restconfConfiguration,
+                             NetconfConfiguration netconfSBPConfiguration, final boolean registerShutdownHook)
+            throws ConfigurationException, ExecutionException, InterruptedException, TimeoutException {
 
         //1. initialize and start Lighty controller (MD-SAL, Controller, YangTools, Akka)
         LOG.info("Trying to start LightyController ...");
         LightyControllerBuilder lightyControllerBuilder = new LightyControllerBuilder();
-        LightyController lightyController = lightyControllerBuilder.from(controllerConfiguration).build();
-        lightyController.start().get();
+        this.lightyController = lightyControllerBuilder.from(controllerConfiguration).build();
+        final boolean controllerStartOk = this.lightyController.start().get(DEFAULT_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+        if (!controllerStartOk) {
+            throw new InterruptedException("Lighty.io Controller startup failed!");
```

Comment:
This is not a correct use of InterruptedException. Fix it in all places in your changes.

Reply:
made custom ModuleStartupException and replaced the InterruptedException with that one